snuba/utils/streams/metrics.py

```diff
@@ -1,10 +1,11 @@
-from abc import ABC, abstractmethod
-from typing import Any, Mapping, Optional, Union
+from abc import abstractmethod
+from typing import Any, Mapping, Optional, Protocol, Union, runtime_checkable
 
 Tags = Optional[Mapping[str, str]]
 
 
-class Metrics(ABC):
+@runtime_checkable
+class Metrics(Protocol):
```

Comment:
I guess you turned this into a protocol to be able to depend on this interface without the sentry and Snuba metric classes to extend the same abstract class. Is this right? If yes, great idea.